### PR TITLE
Kogito 2275 - trusty service uses persistence abstraction

### PR DESCRIPTION
**Pull request by r00ta**
Jira link: https://issues.redhat.com/browse/KOGITO-2275

The aim of this PR is to use the persistence abstraction in the trusty service. In addition to that, the endpoint `v1/executions/decisions/{executionId}` is implemented and tested.
The class `InfinispanServerTestResource` has been moved to the `persistence-commons-infinispan` test module, so that any module that would like to use the infinispan testcontainer image can extend it and define what are the caches that have to be erased after each test. 

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [v ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [v ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ v] Pull Request contains link to the JIRA issue
- [ v] Pull Request contains link to any dependent or related Pull Request
- [ v] Pull Request contains description of the issue
- [ v] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: trusty/trusty-service/src/main/resources/META-INF/kogito-cache-default.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (danielezonca) [inline trusty/trusty-service/src/main/resources/META-INF/kogito-cache-default.xml]: Is this file needed?

**Comment by r00ta** [comment]
jenkins retest this

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+@ApplicationScoped
+public class ProtostreamProducer {
+
+    @Inject
+    ObjectMapper mapper;
+
+    @Produces
+    FileDescriptorSource kogitoTypesDescriptor() throws IOException {
+        FileDescriptorSource source = new FileDescriptorSource();
+        source.addProtoFile("decision.proto", Thread.currentThread().getContextClassLoader().getResourceAsStream("META-INF/decision.proto"));
```
Comment (jiripetrlik) [inline trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java:39]: Is there any reason to use context classloader "Thread.currentThread().getContextClassLoader()" instead of "this.getClass().getResourceAsStream()".

File: trusty/trusty-storage/trusty-storage-mongo/pom.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jiripetrlik) [inline trusty/trusty-storage/trusty-storage-mongo/pom.xml]: There is only a pom in this module. I would suggest to create Mongo implementation in a separate PR. By creating an empty Mongo module we probably make a false assumption that we implement it.

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/mocks/StorageImplMock.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
+package org.kie.kogito.trusty.service.mocks;
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/mocks/StorageImplMock.java]: Please add missing license header.

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
+package org.kie.kogito.trusty.storage.infinispan;
```
Comment (jiripetrlik) [inline trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java:17]: Please add missing license header.

File: trusty/trusty-storage/trusty-storage-api/src/main/java/org/kie/kogito/trusty/storage/api/TrustyStorageServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package org.kie.kogito.trusty.storage.api;
```
Comment (jiripetrlik) [inline trusty/trusty-storage/trusty-storage-api/src/main/java/org/kie/kogito/trusty/storage/api/TrustyStorageServiceImpl.java:17]: Missing license header.

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/api/DecisionApiV1.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
+package org.kie.kogito.trusty.service.api;
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/api/DecisionApiV1.java]: Missing license header.

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java [hunk 8]
```diff
@@ -16,6 +16,92 @@
...
 public class TrustyServiceTest {
 
+    @Inject
+    TrustyService trustyService;
+
+    @InjectMock
+    TrustyStorageService storageService;
+
+    @Test
+    void GivenADecision_WhenStoreDecisionIsCalled_ThenNoExceptionsAreThrown() {
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java]: I would suggest to not use a capital letter as the first letter of the method.  Also do not use underline.

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java [hunk 9]
```diff
@@ -16,6 +16,92 @@
...
+        Decision decision = new Decision();
+        Storage storageMock = mock(Storage.class);
+        when(storageMock.put(any(Object.class), any(Object.class))).thenReturn(decision);
+
+        when(storageService.getDecisionsStorage()).thenReturn(storageMock);
+        Assertions.assertDoesNotThrow(() -> trustyService.storeDecision("test", decision));
+    }
+
+    @Test
+    void GivenADecision_WhenADecisionIsStoredAndRetrieved_ThenTheOriginalObjectIsReturned() {
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java]: I would suggest to not use a capital letter as the first letter of the method. Also do not use underline.

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java [hunk 10]
```diff
@@ -16,6 +16,92 @@
...
+        trustyService.storeDecision("executionId", decision);
+
+        List<Execution> result = trustyService.getExecutionHeaders(OffsetDateTime.now().minusDays(1), OffsetDateTime.now(), 100, 0, "");
+
+        Assertions.assertEquals(1, result.size());
+        Assertions.assertEquals(decision.getExecutionId(), result.get(0).getExecutionId());
+    }
+
+    @Test
+    void GivenADecision_WhenADecisionIsStoredAndRetrievedById_ThenTheOriginalObjectIsReturned() {
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java]: I would suggest to not use a capital letter as the first letter of the method.  Also do not use underline.

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceIT.java [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+        OffsetDateTime from = OffsetDateTime.ofInstant(Instant.ofEpochMilli(1591692940000L), ZoneOffset.UTC);
+        OffsetDateTime to = OffsetDateTime.ofInstant(Instant.ofEpochMilli(1591692959000L), ZoneOffset.UTC);
+        List<Execution> result = trustyService.getExecutionHeaders(from, to, 100, 0, "my");
+        Assertions.assertEquals(1, result.size());
+        Assertions.assertEquals("myExecution", result.get(0).getExecutionId());
+    }
+
+    @Test
+    public void GivenAnExecution_WhenGetDecisionByIdIsCalled_ThenTheExecutionIsReturned() {
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceIT.java]: I would suggest to not use a capital letter as the first letter of the method. Also do not use underline.

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceIT.java [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+        String executionId = "myExecution";
+        storeExecution(executionId, 1591692950000L);
+
+        Decision result = trustyService.getDecisionById(executionId);
+        Assertions.assertEquals(executionId, result.getExecutionId());
+    }
+
+    @Test
+    public void GivenADuplicatedDecision_WhenTheDecisionIsStored_ThenAnExceptionIsRaised() {
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceIT.java]: I would suggest to not use a capital letter as the first letter of the method. Also do not use underline.

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceIT.java [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+        String executionId = "myExecution";
+        storeExecution(executionId, 1591692950000L);
+        Assertions.assertThrows(IllegalArgumentException.class, () -> storeExecution(executionId, 1591692950000L));
+    }
+
+    @Test
+    public void GivenNoExecutions_WhenADecisionIsRetrieved_ThenAnExceptionIsRaised() {
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceIT.java]: I would suggest to not use a capital letter as the first letter of the method. Also do not use underline.

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceIT.java [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+
+        OffsetDateTime from = OffsetDateTime.ofInstant(Instant.ofEpochMilli(1591692957000L), ZoneOffset.UTC);
+        OffsetDateTime to = OffsetDateTime.ofInstant(Instant.ofEpochMilli(1591692959000L), ZoneOffset.UTC);
+        List<Execution> result = trustyService.getExecutionHeaders(from, to, 100, 0, "");
+        Assertions.assertEquals(1, result.size());
+        Assertions.assertEquals("myExecution", result.get(0).getExecutionId());
+    }
+
+    @Test
+    public void GivenTwoExecutions_WhenTheQueryExcludesOneExecution_ThenOnlyOneExecutionIsReturned() {
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceIT.java]: I would suggest to not use a capital letter as the first letter of the method. Also do not use underline.

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceIT.java [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+
+        OffsetDateTime from = OffsetDateTime.ofInstant(Instant.ofEpochMilli(1591692940000L), ZoneOffset.UTC);
+        OffsetDateTime to = OffsetDateTime.ofInstant(Instant.ofEpochMilli(1591692955000L), ZoneOffset.UTC);
+        List<Execution> result = trustyService.getExecutionHeaders(from, to, 100, 0, "");
+        Assertions.assertEquals(1, result.size());
+        Assertions.assertEquals("myExecution", result.get(0).getExecutionId());
+    }
+
+    @Test
+    public void GivenTwoExecutions_WhenThePrefixIsUsed_ThenOnlyOneExecutionIsReturned() {
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceIT.java]: I would suggest to not use a capital letter as the first letter of the method. Also do not use underline.

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceIT.java [hunk 16]
```diff
@@ -0,0 +1,98 @@
+package org.kie.kogito.trusty.service;
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceIT.java:17]: License header is missing.

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyInfinispanServerTestResource.java [hunk 17]
```diff
@@ -0,0 +1,11 @@
+package org.kie.kogito.trusty.service;
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyInfinispanServerTestResource.java:17]: License header is missing.

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/api/DecisionsApiV1.java [hunk 18]
```diff
@@ -0,0 +1,58 @@
+package org.kie.kogito.trusty.service.api;
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/api/DecisionsApiV1.java:17]: License header is missing.

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/api/DecisionsApiV1.java [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+import org.kie.kogito.trusty.service.ITrustyService;
+import org.kie.kogito.trusty.service.responses.ExecutionHeaderResponse;
+import org.kie.kogito.trusty.storage.api.model.Decision;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("v1/executions/decisions")
+public class DecisionsApiV1 {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DecisionsApiV1.class);
```
Comment (jiripetrlik) [inline trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/api/DecisionsApiV1.java]: It seems that this logger is not used.

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.infinispan.protostream.MessageMarshaller;
+import org.kie.kogito.persistence.infinispan.protostream.AbstractMarshaller;
+import org.kie.kogito.trusty.storage.api.model.Decision;
+import org.kie.kogito.trusty.storage.api.model.ExecutionTypeEnum;
+
+public class DecisionMarshaller extends AbstractMarshaller implements MessageMarshaller<Decision> {
```
Comment (jiripetrlik) [inline trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:28]: MessageMarshaller is deprecated according to Sonar cloud.

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java [hunk 21]
```diff
@@ -0,0 +1,52 @@
+package org.kie.kogito.trusty.storage.infinispan;
```
Comment (jiripetrlik) [inline trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:17]: Missing license header.

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+
+    @Produces
+    FileDescriptorSource kogitoTypesDescriptor() throws IOException {
+        FileDescriptorSource source = new FileDescriptorSource();
+        source.addProtoFile("decision.proto", Thread.currentThread().getContextClassLoader().getResourceAsStream("META-INF/decision.proto"));
+        return source;
+    }
+
+    @Produces
+    MessageMarshaller decisionMarshaller() {
```
Comment (jiripetrlik) [inline trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java:44]: MessageMarshaller is deprecated according to Sonar cloud.

**Review by jiripetrlik — CHANGES_REQUESTED** [review]

File: trusty/trusty-storage/trusty-storage-mongo/pom.xml (hunk 3 above)
Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-mongo/pom.xml]: Done

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java (hunk 2 above)
Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java:39]: Hi, the equivalent code would be `this.getClass().getClassLoader().getResourceAsStream("META-INF/decision.proto"))`. TCCL is in general preferred, but in this case it's not a big deal (https://access.redhat.com/solutions/3097731). lmkwyt

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java (hunk 20 above)
Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:28]: Yeah, I know. Currently the documentation of infinispan still uses it (https://infinispan.org/docs/dev/titles/developing/developing.html), and this is currently what we do in the data-index (https://github.com/kiegroup/kogito-apps/search?q=MessageMarshaller&unscoped_q=MessageMarshaller). I'll look for an alternative

Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:28]: Apparently this is the only method to create a custom marshaller when it is not possible to annotate the class that has to be stored (in this case `Decision`). As a matter of facts, the idea is that `Decision` is in the `api` module and it is not infinispan specific. I'll ping the infinispan team to understand if there is an alternative

Screenshot from 2020-06-22 10-45-47

Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:28]: Unfortunately, this feature still has to be implemented https://issues.redhat.com/browse/IPROTO-137 . lmkwyt @jiripetrlik

Comment (jiripetrlik) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:28]: @r00ta 
Hello, thank you for quick reply! I think we can keep deprecated interface for now and comment it with TODO and link to the https://issues.redhat.com/browse/IPROTO-137. Once IPROTO-137 is finished we can decide what to do about it.

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java (hunk 2 above)
Comment (jiripetrlik) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java:39]: @r00ta 
I agree that this is a detail. But if I understand correctly the code we want to load "META-INF/decision.proto" is in the same jar as class and we do not expect that user will change this file in his deployment. Therefore I expect we want to use the same classloader as our class do.

However, I understand that this is only a detail and we will probably never do Wildfly module or something from this code. So, I have no strong opinion on this.

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyService.java [hunk 23]
```diff
@@ -17,25 +17,50 @@
...
+    TrustyStorageService storageService;
+
     @Override
     public List<Execution> getExecutionHeaders(OffsetDateTime from, OffsetDateTime to, int limit, int offset, String prefix) {
-        throw new NotImplementedYetException("Not implemented yet.");
+        Storage<String, Decision> storage = storageService.getDecisionsStorage();
+        List<AttributeFilter<?>> filters = new ArrayList<>();
+        filters.add(QueryFilterFactory.like("executionId", prefix + "*"));
+        filters.add(QueryFilterFactory.greaterThanEqual("executionTimestamp", from.toInstant().toEpochMilli()));
+        filters.add(QueryFilterFactory.lessThanEqual("executionTimestamp", to.toInstant().toEpochMilli()));
```
Comment (danielezonca) [inline trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyService.java]: Can you move strings to a constant class?

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyService.java [hunk 24]
```diff
@@ -17,25 +17,50 @@
...
     }
 
     @Override
-    public void storeExecution(String executionId, Execution execution) {
-        throw new NotImplementedYetException("Not implemented yet.");
+    public void storeDecision(String executionId, Decision decision) {
+        Storage<String, Decision> storage = storageService.getDecisionsStorage();
+        if (storage.containsKey(executionId)) {
+            throw new IllegalArgumentException(String.format("A decision with ID %s is already present in the storage.", executionId));
+        }
```
Comment (danielezonca) [inline trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyService.java:71]: Are you sure of this behavior? Don't we want to support update?

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/api/DecisionsApiV1.java [hunk 25]
```diff
@@ -0,0 +1,70 @@
...
+                    description = "The execution ID.",
+                    required = true,
+                    schema = @Schema(implementation = String.class)
+            ) @PathParam("executionId") String executionId) {
+
+        Decision decision;
+        try {
+            decision = trustyService.getDecisionById(executionId);
+        } catch (IllegalArgumentException ex) {
+            return Response.status(Response.Status.BAD_REQUEST.getStatusCode(), String.format("Execution with ID %s does not exist in the storage.", executionId)).build();
```
Comment (danielezonca) [inline trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/api/DecisionsApiV1.java]: I would prefer to get error message from `ex.getMessage()` otherwise it is possible to "hide" the real error message with this new string

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java [hunk 26]
```diff
@@ -16,6 +16,87 @@
...
+import org.kie.kogito.trusty.storage.api.TrustyStorageService;
+import org.kie.kogito.trusty.storage.api.model.Decision;
+import org.kie.kogito.trusty.storage.api.model.Execution;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@QuarkusTest
```
Comment (danielezonca) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java]: If you are not using REST api nor using many injections, you can probably use a plain unit test without QuarkusTest.
I would suggest you to create a constructor in `TrustyService` that accept `TrustyStorageService` and move `@Inject` annotation directly on the constructor.
In this way you can easily instantiate the service in a `@Before` method using a mock, a mocked implementation or the real instance with better control.
Wdyt?

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java [hunk 27]
```diff
@@ -16,6 +16,87 @@
...
 public class TrustyServiceTest {
```
Comment (danielezonca) [inline trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java:37]: Side note related to this class: if you really want to test `TrustyStorageService` and `TrustyService` in two different classes you should try to properly split test logic: most (all?) the logic is inside `TrustyStorageService` so here you should probably just verify (using Mockito) that method A is invoking method B etc.
The risk is to test the implementation and/or test the mock logic you just defined.
I think it could be better to have a single test class to cover the logic of both. Wdyt?

File: trusty/trusty-service/src/test/resources/application.properties [hunk 28]
```diff
@@ -0,0 +1,10 @@
+# Infinispan
```
Comment (danielezonca) [inline trusty/trusty-service/src/test/resources/application.properties:1]: Is this needed? Are we using infinispan in any test?

File: trusty/trusty-storage/trusty-storage-api/pom.xml [hunk 29]
```diff
@@ -0,0 +1,30 @@
...
+      <artifactId>persistence-commons-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-databind</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-annotations</artifactId>
+    </dependency>
```
Comment (danielezonca) [inline trusty/trusty-storage/trusty-storage-api/pom.xml]: Are them used?

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java [hunk 30]
```diff
@@ -0,0 +1,68 @@
...
+public class DecisionMarshaller extends AbstractMarshaller implements MessageMarshaller<Decision> {
+
+    private static final ObjectMapper myMapper = new ObjectMapper();
+
+    public DecisionMarshaller(ObjectMapper mapper) {
+        super(mapper);
+    }
+
+    @Override
+    public Decision readFrom(ProtoStreamReader reader) throws IOException {
```
Comment (danielezonca) [inline trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:37]: Can you please add a test to verify that all fields of `Decision` are covered (if possible use a mechanism that dynamically takes methods from `Decision` so that if we add a field the test fails

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java [hunk 31]
```diff
@@ -0,0 +1,68 @@
...
+        result.setSuccess(reader.readBoolean("hasSucceeded"));
+        result.setExecutorName(reader.readString("executorName"));
+        result.setExecutedModelName(reader.readString("executedModelName"));
+        result.setExecutionType(myMapper.readValue(reader.readString("executionType"), ExecutionTypeEnum.class));
+
+        return result;
+    }
+
+    @Override
+    public void writeTo(ProtoStreamWriter writer, Decision result) throws IOException {
```
Comment (danielezonca) [inline trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:50]: Same as above

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java (hunk 2 above)
Comment (danielezonca) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java:39]: For the record, fine for me to review this usage but if we decide to change, please update this too https://github.com/kiegroup/kogito-apps/blob/360478879e2d8086707c8d9ee3498883021c8595/data-index/data-index-storage/data-index-storage-protobuf/src/main/java/org/kie/kogito/index/protobuf/ProtostreamProducer.java#L35-L36

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java (hunk 20 above)
Comment (danielezonca) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:28]: @r00ta 
Can you please create a KOGITO ticket (unassigned is fine) to remove the usages of this method so we can track?

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyService.java (hunk 24 above)
Comment (r00ta) [reply trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyService.java:71]: @danielezonca Personally, for the update operation I would create another method (when it will be needed). I think we should not overwrite items in the storage by default, lmkwyt

File: trusty/trusty-service/src/main/resources/META-INF/kogito-cache-default.xml (hunk 1 above)
Comment (r00ta) [reply trusty/trusty-service/src/main/resources/META-INF/kogito-cache-default.xml]: Yes, it is needed by https://github.com/kiegroup/kogito-apps/blob/a57350f0ea61cde20a68f1a2e0fc63dde5922a36/persistence-commons/persistence-commons-infinispan/src/main/java/org/kie/kogito/persistence/infinispan/cache/InfinispanCacheManager.java#L97 . But actually it would make sense to move it to `trusty-persistence-infinispan`, wdyt?

File: trusty/trusty-service/src/test/resources/application.properties (hunk 28 above)
Comment (r00ta) [reply trusty/trusty-service/src/test/resources/application.properties:1]: @danielezonca yes, in order to select which storage layer is to be used, we have to set the properties here. For me the question is how do we write integration tests for mongo for example. I think that at one point we will have to move (some) integration tests out of this module to a separated module (or put them in `kogito-examples`?)

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java (hunk 27 above)
Comment (r00ta) [reply trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java:37]: Currently `TrustyStorageService` is not really tested anywhere, since it is only exposing the "cache" of the decisions. I think the `TrustyStorageService` should be mocked anyway in the tests of the `TrustyService`. I know there is not much to test, as a matter of facts I've implemented just 3 unit tests and some integration tests for the `TrustyService`. Let's have a call to better clarify your points

File: trusty/trusty-storage/trusty-storage-api/pom.xml (hunk 29 above)
Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-api/pom.xml]: Nope, I'll remove them

File: trusty/trusty-service/src/main/resources/META-INF/kogito-cache-default.xml (hunk 1 above)
Comment (danielezonca) [reply trusty/trusty-service/src/main/resources/META-INF/kogito-cache-default.xml]: Yes it should be better :+1:

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/api/DecisionsApiV1.java (hunk 25 above)
Comment (r00ta) [reply trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/api/DecisionsApiV1.java]: Done

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyService.java (hunk 23 above)
Comment (r00ta) [reply trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyService.java]: Done

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java (hunk 26 above)
Comment (r00ta) [reply trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java]: I've avoided `QuarkusTest` now, let me know if it looks good to you

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java (hunk 30 above)
Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:37]: Done, lmkwyt

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java (hunk 31 above)
Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:50]: Done, lmkwyt

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java (hunk 20 above)
Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:28]: @danielezonca @jiripetrlik here it is https://issues.redhat.com/browse/KOGITO-2517

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java (hunk 2 above)
Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java:39]: ok, If you agree I would change it for trusty in the current PR, and then in another one I'll change it for the data-index

Comment (danielezonca) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java:39]: To be honest it is fine for me to stay with the current code, I just mentioned that the same strategy is used in data-index so if we will change, we should update that code too. :)

File: trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyService.java (hunk 24 above)
Comment (r00ta) [reply trusty/trusty-service/src/main/java/org/kie/kogito/trusty/service/TrustyService.java:71]: I've implemented the `updateDecision` method as agreeded by phone call

File: trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java (hunk 26 above)
Comment (r00ta) [reply trusty/trusty-service/src/test/java/org/kie/kogito/trusty/service/TrustyServiceTest.java]: Implemented as agreeded by phone call

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java (hunk 30 above)
Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:37]: The check on the mapping has been implemented using reflection, I'll resolve the conversation

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java (hunk 31 above)
Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/DecisionMarshaller.java:50]: Implemented as agreeded by phone call

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java (hunk 2 above)
Comment (r00ta) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java:39]: Also from my side I think we can stay with the current implementation.

**Comment by r00ta** [comment]
The pull request on kogito-runtimes to add the modules to the bom is here https://github.com/kiegroup/kogito-runtimes/pull/561/files

**Review by danielezonca — APPROVED** [review]
Thanks for you changes, approving :+1:

File: trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java (hunk 2 above)
Comment (jiripetrlik) [reply trusty/trusty-storage/trusty-storage-infinispan/src/main/java/org/kie/kogito/trusty/storage/infinispan/ProtostreamProducer.java:39]: Ok, I have no problem with it.

**Review by jiripetrlik — APPROVED** [review]
@r00ta Thank you for changes! Looks good to me now.
